.github/workflows/linux.yml

```diff
@@ -387,3 +387,48 @@ jobs:
         python -m pip install numpy==1.23.5 cython pybind11 pytest pytest-timeout pytest-xdist pytest-env 'Pillow<10.0.0' mpmath pythran pooch meson hypothesis && \
         LD_LIBRARY_PATH=/usr/local/lib python dev.py build && \
         LD_LIBRARY_PATH=/usr/local/lib python dev.py test"
+
+  #################################################################################
+  distro_multiple_pythons:
+    # Purpose is to build for a non-default Python interpreter in a Linux distro
+    # For such a build config, `python`/`python3` executables may not have
+    # build dependencies like Cython or NumPy installed.
+    name: non-default Python interpreter, fast, py3.10/npMin, pip+pytest
+    needs: get_commit_message
+    if: >
+      needs.get_commit_message.outputs.message == 1
+      && (github.repository == 'scipy/scipy' || github.repository == '')
+    runs-on: ubuntu-22.04
+    steps:
+      - uses: actions/checkout@v4.1.1
+        with:
+          submodules: recursive
+
+      - name: Setup system dependencies
+        run: |
+          sudo apt-get -y update
+          # `python3-dev` yields Python 3.10 on Ubuntu 22.04
+          sudo apt install -y python3-dev python3.11-dev ninja-build pkg-config libatlas-base-dev liblapack-dev
+
+      - name: Setup Python build deps
+        run: |
+          python3.11 -m pip install build pythran pybind11 cython numpy meson-python
+
+      - name: Build wheel and install
+        run: |
+          python3.11 -m build -wnx -Csetup-args=-Dblas=blas-atlas -Csetup-args=-Dlapack=lapack-atlas -Ccompile-args=-j2
```

Comment:
Are we using `build -x` because `pyproject.toml` requires `numpy>=2.0.0rc1`?

Reply:
Yes indeed. `-n` and `-x` almost always go together, because usually you're skipping build isolation to target an environment that is somehow different from what you'd get from build isolation.